lib/integrations/externalcloudaudit/bootstrap_test.go

```diff
@@ -0,0 +1,232 @@
+package externalcloudaudit
+
+import (
+	"context"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/athena"
+	"github.com/aws/aws-sdk-go-v2/service/glue"
+	gluetypes "github.com/aws/aws-sdk-go-v2/service/glue/types"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	s3types "github.com/aws/aws-sdk-go-v2/service/s3/types"
+	ecatypes "github.com/gravitational/teleport/api/types/externalcloudaudit"
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBootstrapInfra(t *testing.T) {
+	tt := []struct {
+		desc            string
+		region          string
+		eca             *ecatypes.ExternalCloudAuditSpec
+		bucketExists    bool
+		tableExists     bool
+		databaseExists  bool
+		workgroupExists bool
+		errExpected     bool
+	}{
+		{
+			desc:        "nil input",
+			errExpected: true,
+		},
+		{
+			desc:        "standard input",
+			errExpected: false,
+			region:      "us-west-2",
+			eca: &ecatypes.ExternalCloudAuditSpec{
+				SessionsRecordingsURI:  "s3://long-term-storage-bucket",
```

Comment:
I'd include bucket prefixes in most of these testcases since we will be using those